### PR TITLE
nvdimm_mode: Add new case to test change nvdimm mode inside guest

### PR DESCRIPTION
**Pull request by huangyum**
ID: 1843844
Signed-off-by: Yumei Huang <yuhuang@redhat.com>
Depends on https://github.com/avocado-framework/avocado-vt/pull/2585

### CODE REVIEW

**Comment by huangyum** [comment]
(1/1) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_mode: PASS (59.98 s)

@PaulYuuu @yanan-fu Please help review, thanks.

File: qemu/tests/cfg/nvdimm.cfg [hunk 1]
```diff
@@ -66,3 +66,19 @@
...
+        - nvdimm_mode:
+            type = nvdimm_mode
+            mem_devs = mem1 mem2
+            mem-path_mem1 = /tmp/test1.img
+            mem-path_mem2 = /tmp/test2.img
+            pre_command = "dd if=/dev/zero of=${mem-path_mem1} bs=1M count=1024"
+            pre_command += "; dd if=/dev/zero of=${mem-path_mem2} bs=1M count=1024"
+            post_command = "rm -rf ${mem-path_mem1}"
+            post_command += "; rm -rf ${mem-path_mem2}"
+            ndctl_install_cmd = "rpm -q ndctl || yum install -y ndctl"
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/cfg/nvdimm.cfg]: Updated, please help review, thanks.

File: qemu/tests/cfg/nvdimm.cfg [hunk 2]
```diff
@@ -66,3 +66,20 @@
             check_command = "grep -A21 '${nv_backend}' /proc/%s/smaps"
             clean_command = "rm -rf ${mount_dir}/${nv_backend} && umount ${mount_dir} && rm -rf ${mount_dir}"
             start_vm = no
+        - nvdimm_mode:
+            type = nvdimm_mode
+            mem_devs = mem1 mem2
+            mem-path_mem1 = /tmp/test1.img
+            mem-path_mem2 = /tmp/test2.img
+            align_mem1 = 128M
+            align_mem2 = 128M
```
Comment (yanan-fu) [inline qemu/tests/cfg/nvdimm.cfg:75]: Ah~  you add it now.

And, how about `label-size` ?

Comment (huangyum) [reply qemu/tests/cfg/nvdimm.cfg:75]: It's in the next line.

Comment (yanan-fu) [reply qemu/tests/cfg/nvdimm.cfg:75]: Yeap,  i miss it. thanks!

**Comment by PaulYuuu** [comment]
Since nvdimm device on powerpc still has a product bug currently, I will verify this case can work on powerpc when the device available.

Ack.

**Comment by huangyum** [comment]
@yanan-fu Any more comments?

File: qemu/tests/nvdimm_mode.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    utils_package.package_install("ndctl", session)
+    create_dax_cmd = params["create_dax_cmd"]
+    nvdimm_number = len(params["mem_devs"].split())
+    try:
+        for i in range(nvdimm_number):
+            session.cmd(create_dax_cmd % i)
+        output = session.cmd_output(params["ndctl_check_cmd"])
+        dax_number = len(re.findall('"mode":"devdax"', output))
```
Comment (yanan-fu) [inline qemu/tests/nvdimm_mode.py:29]: I prefer to check the key-value,more robust. 
```
output = eval(output)
for item in output:
    if item['mode'] != 'devdax':
        test.fail(****)     
```

It is up to you.

**Comment by yanan-fu** [comment]
I am not familiar with the `align`.  When set 128M in the command line, the output of `ndctl list` show it is `4096` . Is it normal ?

File: qemu/tests/nvdimm_mode.py (hunk 3 above)
Comment (huangyum) [reply qemu/tests/nvdimm_mode.py:29]: > I am not familiar with the `align`. When set 128M in the command line, the output of `ndctl list` show it is `4096` . Is it normal ?

Yes, they have different meaning. The align inside guest should be the same value in 
`create_dax_cmd = "ndctl create-namespace -m dax -e namespace%s.0 -f -v -a 8192"`

File: qemu/tests/nvdimm_mode.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    3) Check if both devices are dax mode
+
+    :param test: QEMU test object
+    :param params: Dictionary with the test parameters
+    :param env: Dictionary with test environment.
+    """
+    vm = env.get_vm(params["main_vm"])
+    vm.verify_alive()
+    session = vm.wait_for_login()
+    utils_package.package_install("ndctl", session)
```
[the remark this answers is not in the document]
Comment (huangyum) [reply qemu/tests/nvdimm_mode.py]: Updated, pls review again, thanks.

**Comment by huangyum** [comment]
(1/1) Host_RHEL.m8.u3.product_rhel.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.8.3.0.x86_64.io-github-autotest-qemu.nvdimm.nvdimm_mode: PASS (93.28 s)

**Pull request merged**